### PR TITLE
Scenario objective fixes

### PR DESCRIPTION
**Pull request by NickAragua**
Changes relevant to #1343 and updates to scenario template definitions.

Requires companion megamek PR.

### CODE REVIEW

**Comment by NickAragua** [comment]
Ok, I think this is ready to be reviewed. Basically it addresses a bunch of issues that people found when I messed around with AtB scenario briefings, and also a bunch of issues relating to scenario templates and modifiers.

File: MekHQ/data/scenariotemplates/Recon.xml [hunk 1]
```diff
@@ -104,8 +106,11 @@
                     <howMuch>1</howMuch>
                 </failureEffect>
             </failureEffects>
-            <additionalDetails/>
-            <description>Scan the following enemy unit(s) and force(s). Scans may be conducted by weapons fire, TAG, spotting or physical attacks. If the scan action requires a roll, it must be possible:</description>
+            <additionalDetails>
+                <additionalDetail>Scans may be conducted by weapons fire, TAG, spotting or physical attacks.</additionalDetail>
+                <additionalDetail>If the scan action requires a roll, it must be possible.</additionalDetail>
```
Comment (sixlettervariables) [inline MekHQ/data/scenariotemplates/Recon.xml]: The old text ended in a colon, was that unintentional?

File: MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioModifier.java [hunk 2]
```diff
@@ -255,7 +253,7 @@ public static AtBScenarioModifier Deserialize(String fileName) {
                 resultingList = templateElement.getValue();
             }
         } catch(Exception e) {
-            MekHQ.getLogger().error(ScenarioModifierManifest.class, "Deserialize", "Error Deserializing Scenario Modifier", e);
+            MekHQ.getLogger().error(ScenarioModifierManifest.class, "Deserialize", "Error Deserializing Scenario Modifier" + fileName, e);
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioModifier.java]: Needs a colon and a space before the filename.

File: MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioModifierApplicator.java [hunk 3]
```diff
@@ -143,18 +143,9 @@ public static void inflictBattleDamage(AtBDynamicScenario scenario, Campaign cam
                     
                     for(int clusterCount = 0; clusterCount < numClusters; clusterCount++) {
                         HitData hitData = en.rollHitLocation(ToHitData.HIT_NORMAL, Compute.randomInt(4));
-                        int resultingArmor = en.getArmor(hitData) - 5;
+                        int resultingArmor = Math.max(1, en.getArmor(hitData) - 5);
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioModifierApplicator.java]: So this is no longer removing internals?

Comment (NickAragua) [reply MekHQ/src/mekhq/campaign/mission/atb/AtBScenarioModifierApplicator.java]: Correct. I'm also having some difficulty getting it to actually update the MegaMek entity when it gets sent over, but I'll address that in another PR, this one is getting a little large.

File: MekHQ/data/scenariotemplates/Recon.xml (hunk 1 above)
Comment (NickAragua) [reply MekHQ/data/scenariotemplates/Recon.xml]: Probably not, let me double check.

Comment (NickAragua) [reply MekHQ/data/scenariotemplates/Recon.xml]: Ah, ok. The serializer alphabetized the property names. If you look at "description", it still has the colon at the end. It looks weird on the scenario resolution screen when the description is too long for a custom objective type, so I solved the problem by shortening the description. :D

**Pull request merged**